updating package.json

Kflash opened the pull request:
- updated package.json
- renamed name from ES6 to ES2015
- added the beginning of a Eslint integration

Kflash commented:
@jmeas Eslint I just added it to package.json, and the script section. No idea what kind of rules you need, but most of the things are setup if you add a .eslintsrc file in the root. I guess you are not using Sublime?

Kflash commented on package.json:
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "test": "gulp",
     "test-browser": "gulp test-browser",
+    "lint": "./node_modules/.bin/eslint src",
```
This has to be changed to match your current setup.

jmeas commented:
Thanks for the PR, @kflash. It seems like this one commit is doing three different things.

1. updating the language from ES6 to ES2015
2. updating the dependencies
3. beginning to add eslint, but not finishing the job

I think that each of these merit their own commits, and possibly even PRs, just so that no single one holds up the others. With that said, I have a few preliminary comments on each change:

#### One

I'd probably merge this if you made a separate PR. With that said, I'm pretty sure the name ES6 will continue to stick around in the lexicon, even if it's not Officially Correct©. So I'm not *too* concerned about using the name ES6. :stuck_out_tongue: 

#### Two

Awesome, sure. Can you make that a separate PR/commit?

#### Three

I only want to add ESLint if JSHint is completely removed. Including both in the repo is kinda silly, I think.

jmeas commented on package.json:
```diff
@@ -30,36 +31,39 @@
-    "babelify": "^6.0.0",
-    "browserify": "^8.1.1",
-    "chai": "^2.0.0",
+    "babel": "^5.2.17",
```
Some of these changes aren't actually doing anything. The `^` operator will pull in minor and patch releases. So, for instance, `^5.0.0` will pull in any version of Babel that is less than `^6.0.0`, including `^5.2.17`.

Kflash replied:
I know this. But as I stated earlier I get trouble with changing version numbers on all of the dependencies because of nodejJS 12.2. It's require a lot of different versions. At least in my end. So I skipped that part.

For the main files as you pointed out I did this just to try to be compatible with all nodeJS versions. As you are running v. 10 and v. 12 in test cases. But I will remove the operator.

jmeas replied:
> But as I stated earlier I get trouble with changing version numbers on all of the dependencies because of nodejJS 12.2. It's require a lot of different versions. At least in my end. So I skipped that part.

Ah, makes sense. I really shouldn't have assumed that you weren't familiar with the `^` operator – that's my bad! Definitely make whatever changes are necessary to get the build to pass! :)

Kflash commented:
@jmeas I'm on my way out the door, but will fix this as soon as I'm back. And split it into two separate PR.

The majority of people - including myself - are using the name ES6, so agree there. But at least pull it in as a reference maybe? I only changed the readme, and as you may note I added in parentheses *( prev: ES6)*

For the ESLint there would be a different task if you are using sublime. To integrate ESLint into the project itself and remove the other linting tools is easy, but little more tricky when you use Sublime.

I will take a look when I'm back home.

Kflash commented:
@jmeas There is something you need to consider. I'm closing this pull and will send the others shortly. What you need to consider is which exact versions works smooth with nodejS v.10 and v.12 and not and same for iosjs. Then you put up a hook for this in the TravicCI as one example.

If you look at the jQuery repo, and their package.json: https://github.com/jquery/jquery/blob/master/package.json#L56-L59

Here you see that they setting up different jsDOM for ios and nodejs. To get a bullet proof solution that works for everyone, you should cross check to see if all Babel versions works with nodeJS v.10 and if not use the old version that works. 

I think you got my idea. And if you google you see that nodejs v.12 is not 100% stable.

Kflash commented:
@jmeas Only package.json change in my end now. But it seems that your readme will miss some text. No idea why, but easy to add back. 

For EsLint and name change I'm awaiting your conclusion

Kflash commented:
@jmeas For the **uglifyJS** that throws a warning on nodeJS v. 12 in your setup, you can change it to this:

**.pipe($.uglify({ preserveComments: "some"}))**

I'm re-factoring my own build system, therefor I know. Even if you do this, you will get a sourcemap.

Kflash commented:
@jmeas It's up to you. If this get pulled it will only fix the dependencies and revoke my oldest changes to your current setup.

I will ASAP try to fix ESLint for you too, but had to sort out another issue for you first.

jmeas commented on README.md:
```diff
@@ -70,8 +70,8 @@ and your tests.
-You're authoring any library that exports a single file. Examples are small libraries,
-or even entire JavaScript web applications. I use this boilerplate for both!
+You're authoring a library that exports a single file, and that one file
+exports a single variable.
```
Looks like you need to rebase off master.

jmeas commented:
@kflash, thank you so much for this PR! You really motivated me to get these changes in.

Over time, these changes have eventually landed into master, so I'm going to close this PR.

THANK YOU AGAIN :tada:
